### ISSUE TITLE
Readable Stream not being identified as iterable in CommonJS

### ISSUE THREAD

**Issue by Marcos-Costa** [microsoft/TypeScript#39051]
**TypeScript Version:**  4.0.0-dev.20200611

**Search Terms:**
Async Iterator, Symbol, Readable Stream, CommonJS

**Expected behavior:**

Typescript should not give any errors since [Readable Streams in node supports async iterators](https://nodejs.org/api/stream.html#stream_readable_symbol_asynciterator). 

**Actual behavior:**

```
Type 'ReadableStream<any>' must have a '[Symbol.asyncIterator]()' method that returns an async iterator.
```

**Related Issues:**

**Code**
```ts
export async function foo (file: ReadableStream) {
    for await (const chunk of file) {
        console.log(chunk)
    }
}
```
**Workaround**
```ts
export async function foo (file: Iterable<ReadableStream>) {
    for await (const chunk of file) {
        console.log(chunk)
    }
}
```

<details><summary><b>Output</b></summary>

```ts
var __asyncValues = (this && this.__asyncValues) || function (o) {
    if (!Symbol.asyncIterator) throw new TypeError("Symbol.asyncIterator is not defined.");
    var m = o[Symbol.asyncIterator], i;
    return m ? m.call(o) : (o = typeof __values === "function" ? __values(o) : o[Symbol.iterator](), i = {}, verb("next"), verb("throw"), verb("return"), i[Symbol.asyncIterator] = function () { return this; }, i);
    function verb(n) { i[n] = o[n] && function (v) { return new Promise(function (resolve, reject) { v = o[n](v), settle(resolve, reject, v.done, v.value); }); }; }
    function settle(resolve, reject, d, v) { Promise.resolve(v).then(function(v) { resolve({ value: v, done: d }); }, reject); }
};
export async function foo(file) {
    var e_1, _a;
    try {
        for (var file_1 = __asyncValues(file), file_1_1; file_1_1 = await file_1.next(), !file_1_1.done;) {
            const chunk = file_1_1.value;
            console.log(chunk);
        }
    }
    catch (e_1_1) { e_1 = { error: e_1_1 }; }
    finally {
        try {
            if (file_1_1 && !file_1_1.done && (_a = file_1.return)) await _a.call(file_1);
        }
        finally { if (e_1) throw e_1.error; }
    }
}

```

</details>

<details><summary><b>Compiler Options</b></summary>

```json
{
  "compilerOptions": {
    "noImplicitAny": true,
    "strictNullChecks": true,
    "strictFunctionTypes": true,
    "strictPropertyInitialization": true,
    "strictBindCallApply": true,
    "noImplicitThis": true,
    "noImplicitReturns": true,
    "useDefineForClassFields": false,
    "alwaysStrict": true,
    "allowUnreachableCode": false,
    "allowUnusedLabels": false,
    "downlevelIteration": false,
    "noEmitHelpers": false,
    "noLib": false,
    "noStrictGenericChecks": false,
    "noUnusedLocals": false,
    "noUnusedParameters": false,
    "esModuleInterop": true,
    "preserveConstEnums": false,
    "removeComments": false,
    "skipLibCheck": false,
    "checkJs": false,
    "allowJs": false,
    "declaration": true,
    "experimentalDecorators": false,
    "emitDecoratorMetadata": false,
    "target": "ES2017",
    "module": "CommonJS"
  }
}
```

</details>

**Playground Link:** [Provided](https://www.typescriptlang.org/play/?ts=4.0.0-dev.20200611&ssl=1&ssc=1&pln=5&pc=2#code/KYDwDg9gTgLgBAQwM4E8B2BjOAzArpmASwjRwgjgAptCAbYALjgCVgEATBAI3oGUYobALYBKOAG8AUHBlkoiAO4JC8ShhJJ4GABb4A1nAjYcdYGKmzLcdWiQR6AOloQA5mt1o9I6bIC+k3yA)

**Comment by TrySound** [comment]
Was it fixed?

**Comment by limdauto** [comment]
I'm still seeing this error. Could we reopen please?

**Comment by RyanCavanaugh** [comment]
ReadableStream is defined in the Node .d.ts file on DefinitelyTyped; it'd have to be changed there

**Comment by ColemanDunn** [comment]
> ReadableStream is defined in the Node .d.ts file on DefinitelyTyped; it'd have to be changed there

@RyanCavanaugh does this mean this issue should be opened again but elsewhere? The fact that copying and pasting the example found at https://developer.mozilla.org/en-US/docs/Web/API/ReadableStream#async_iteration results in a compilation error seems like a problem.

**Comment by EzraEn1** [comment]
It seems that the problem is being discussed (https://github.com/DefinitelyTyped/DefinitelyTyped/discussions/62651), but there isn't an issue open for it right now.

**Comment by LorenzoBloedow** [comment]
Solution if you're using Node.js:
```ts
// this
let stream: NodeJS.ReadableStream

// instead of this
let stream: ReadableStream
```

**Comment by GustavoOS** [comment]
> ```ts
> let stream: NodeJS.ReadableStream
> ```

How is the import?
